File: Day228_2025/Q228_Leetcode.py
class Solution:
    def maximum69Number(self, num: int) -> int:
        right_dig_count = -1
        dig_count = 0
        temp = num

        while temp > 0:
            dig = temp % 10
            if dig == 6:
                right_dig_count = dig_count
            dig_count += 1
            temp //= 10

        return num + (0 if right_dig_count == -1 else 3 * (10 ** right_dig_count))

File: Day228_2025/Q228_Leetcode.java
class Q228_Leetcode {
    public int maximum69Number(int num) {
        int rightDigCount = -1;
        int digCount = 0;
        int temp = num;

        while (temp > 0) {
            int dig = temp % 10;
            if (dig == 6) rightDigCount = digCount;

            digCount++;
            temp /= 10;
        }

        return num + (rightDigCount == -1 ? 0 : (3 * (int)Math.pow(10, rightDigCount)));
    }
}
